conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala

```diff
@@ -761,11 +761,41 @@ object TezosDatabaseOperations extends LazyLogging {
     * Given the operation kind, return range of fees and timestamp for that operation.
     * @param kind                 Operation kind
     * @param numberOfFeesAveraged How many values to use for statistics computations
+    * @param asOf                 When the computation is to be considered, by default uses the time of invocation
     * @return                     The average fees for a given operation kind, if it exists
     */
-  def calculateAverageFees(kind: String, numberOfFeesAveraged: Int)(
+  def calculateAverageFees(
+      kind: String,
+      numberOfFeesAveraged: Int,
+      asOf: Instant = Instant.now()
+  )(
       implicit ec: ExecutionContext
   ): DBIO[Option[AverageFees]] = {
+    /* We need to limit the past timestamps for this computation to a reasonable value.
+     * Otherwise the query optimizer won't be able to efficiently use the indexing and
+     * will do a full table scan.
+     *
+     * This is what we know now:
+     * - each cycle bakes 4096 blocks
+     * - a cycle takes around 2-3 days to run
+     * - each block stores a variable number of transactions, down-to the min of 1.
+     *
+     * We can make conservative computations to figure out how far in the past we need to go,
+     * to guarantee a [[numberOfFeesAveraged]] values.
+     *
+     * We can assume a single transaction per block (1 trans/block), hence we need numberOfFeesAveraged blocks.
```

Comment:
Assuming a single transaction per block will not be sound in the long term. Is it possible to either:

1. Move the calculation completely to the database?
2. Simply change the function so it takes a time range rather than number of transactions?

Reply:
Let me first put down a couple of details that may help us:

1. for our current fees averaging count configured to 1000, we don't need to get back past 1 day in the worst case
2. Assuming 1 transaction/block means handling the worst possible case, that is, the longest possible time-window to get all the required transaction needed for averaging.

The latter is to say that blocks with a lot more transactions could impact performance negatively, for sure, but never as much as when doing a table scan (required to filter out based on the outcome of the sort operation).
Once we've cut out most results due to the time-window, I'd assume that the db will be more than able to handle a few million records?
This is only theoretical, so we might want to add measurements to this code and have timing results based on the number of records, and decide on those.

That said, we can move the averaging to the db nonetheless.
Also doing the average on time-range is also reasonable, but I don't know if it's still sensible from the "business" perspective, I lack the perspective on that.

Give me some feedback, please based on the additional information @vishakh

Reply:
Your reasoning makes sense to me. However, we effectively end up with a time window. Won't it be simpler to just explicitly make it a time window and use your heuristic to set the value in a config?

Reply:
I see the point, then this is what I plan to do:

1. switch to time-windowing in config (say how many days past should the averaging go)
2. set the config to 1 day for now
3. see how hard it is to compute the averaging directly as part of the db-query

Reply:
Sounds good to me!

Reply:
Part 1 and 2 are done

Looking at 3

Reply:
Completed all the points